js/router/BRouter.js

```diff
@@ -231,6 +236,39 @@ L.BRouter = L.Class.extend({
         return lonlats;
     },
 
+    _getLonLatsNameString: function(latLngNames) {
+        var s = '';
+        for (var i = 0; i < latLngNames.length; i++) {
+            s += this._formatLatLng(latLngNames[i].latlng);
+            s += L.BRouter.NUMBER_SEPARATOR;
+            s += latLngNames[i].name;
```

Comment:
Then `name` part should be encoded with `encodeURIComponent`.

Otherwise e.g. a name like `a&b` will break the parsing later (not showing the second POI and only "a" as name): 
http://localhost:3000/#map=6/50.965/8.053/standard&pois=2.351632,48.850258,a&b;13.338911,52.536273,c

We are not encoding the whole URL for better readability and because we know it is not needed, but we cannot control user input, therefore such parts do need encoding.